README.md
# Direct Link Translator WordPress Plugin

## Description
This plugin is my first project as I needed it for my work. For building an online language learning system, I needed a quick way to let the users directly translate specific words or sentences on translation services within a lection. 
Right now it translates from German to English.   

## Features
You can add the translation tags where shortcodes work. The link is larger than the surrounding text. You can add standard HTML colors (red, orange, blue) or specific one as hex (RGB not tested). For example:
`[DeepL color=#c53222] Das ist ein Beispiel [/DeepL]`. The link on the page for DeepL will look like this: [**Das ist ein Beispiel.** 🌐](https://www.deepl.com/en/translator#de/en/Das%20ist%20ein%20Beispiel.)

The Shortcodes are:

Shortcode | Translator service
------------ | -------------
`[DeepL] ... [/DeepL]` | [DeepL](https://www.deepl.com/)
`[GTranslate] ... [/GTranslate]` | [Google Translate](https://translate.google.com/)
`[Yandex] ... [/Yandex]` | [Yandex Translate](https://translate.yandex.com/)
`[Baidu] ... [/Baidu]` | [Fanyi Baidu](https://fanyi.baidu.com/)

You can also set specific parameters, that differs from the standard setup:
`[DeepL fromLang=en toLang=it color=#735686]`
- The *fromLang* parameter defines the language of the text.
- The *toLang* parameter defines the language to translate to.
- With the color parameter you can set up your own colors (text or Hex-Color-Code)

## Roadmap
- [x] Add globus emoji behind the string
- [x] Add more translation services
- [X] Add the language code of referenced text
- [X] Add destination language code
- [ ] Add a dropdown with some important languages (or all languages, as DeepL only offers a few)
- [ ] Add a standard reference language every text is translated from

## Legal Notices

This plugin uses third party services by offering a direct link to their website. I am not responsible for any errors happening on their website and can not guarantee the plugin to work at all times.

I am not associated with any of these services. The used services are listed below:

### DeepL
[DeepL](https://deepl.com) is an online translator service. On their website, you can find their [Terms and Conditions](https://www.deepl.com/pro-license#free) and their [Privacy Policy](https://www.deepl.com/privacy/).
### Google Translate
[Google Translate](https://translate.google.com/) is an online translator service from Alphabet Inc.. On their website, you can find their [Terms and Conditions and Policies](https://policies.google.com/).
### Yandex.Translate
[Yandex.Translate](https://translate.yandex.com/) is an online translator service offered by YANDEX LLC. On their website, you can find their [Terms of Use](https://yandex.com/legal/translate_termsofuse/).
### Fanyi Baidu
[Fanyi Baidu](https://fanyi.baidu.com/) is an online translator service offered by the chinese search engine Baidu.com. On their website, you can find their [Terms and Conditions](http://www.baidu.com/duty/).


changelog.txt
== Changelog ==

= 0.1.8.1 =
* Released: 22.10.2020
* Fixed wrong title tag

= 0.1.8 =
* Released: 22.10.2020
* Add translatable link name on hover with used service

= 0.1.7 =
* Released: 18.10.2020
* Added legal notices for WordPress 
* Updated Readme

= 0.1.6 =
* Released: 17.10.2020
* Added Google, Yandex and Baidu translation services

= 0.1.5 =
* Released: 16.10.2020
* Initial release of the plugin.


denade-deepl-translate.php
<?php
/**
 * Plugin Name:         DeepL Link Translator
 * Plugin URI:          https://github.com/DJinnzNathan/denade-deepl-translate
 * Description:         Generates a DeepL Translate link 
 * Version:             0.1.5
 * Text Domain:         denade-deepl-translate
 * Author:              Nathan G-Michael
 * Requires at least:   5.2
 * Requires PHP:        7.2
 * License:             GPL v3 or later
 * License URI:         https://www.gnu.org/licenses/gpl-3.0.html
 */
 
 function denade_deepl_translate($atts, $content) {
    $a = shortcode_atts( array(
        'link' => 'https://www.deepl.com/en/translator#de/en/',
        'color' => '#ed9f2d'
        ), $atts );
        $output = '<strong><a href="' . esc_url( $a['link'] ) . esc_attr( $content ) . '" style="font-size: larger; color: ' . esc_attr( $a['color'] ) . ';" target="_blank">' . esc_attr( $content ) . " 🌐" . '</a></strong>';
        return $output;
}

add_shortcode('DeepL', 'denade_deepl_translate');


 


primus-deepl-translate.php
<?php
/**
 * Plugin Name:         DeepL Link Translator
 * Plugin URI:          
 * Description:         Generates a DeepL Translate link 
 * Version:             0.1.5
 * Text Domain:         denade-deepl-translate
 * Author:              Nathan G-Michael
 * Requires at least:   5.2
 * Requires PHP:        7.2
 * License:             GPL v3 or later
 * License URI:         https://www.gnu.org/licenses/gpl-3.0.html
 */
 
 function denade_deepl_translate($atts, $content) {
    $a = shortcode_atts( array(
        'link' => 'https://www.deepl.com/en/translator#de/en/',
        'color' => '#ed9f2d'
        ), $atts );
        $output = '<strong><a href="' . esc_url( $a['link'] ) . esc_attr( $content ) . '" style="font-size: larger; color: ' . esc_attr( $a['color'] ) . ';" target="_blank">' . esc_attr( $content ) . " 🌐" . '</a></strong>';
        return $output;
}

add_shortcode('DeepL', 'denade_deepl_translate');


 


denade-translate.php
<?php
/**
 * Plugin Name:         Direct Link Translator
 * Plugin URI:          https://github.com/DJinnzNathan/denade-translate
 * Description:         Generates a link to the translation service
 * Version:             0.1.8.1
 * Text Domain:         denade-translate
 * Author:              DeNade Media
 * Author URI:          https://denade-media.com
 * Requires at least:   5.2
 * Requires PHP:        7.2
 * License:             GPL v3 or later
 * License URI:         https://www.gnu.org/licenses/gpl-3.0.html
 */
 
//   DeepL 
 function denade_deepl_translate($atts, $content) {
    $a = shortcode_atts( array(
        'color' => '#ed9f2d',
        'fromLang' => 'de',
        'toLang' => 'en'
        ), $atts );
        $output = '<strong><a href="https://www.deepl.com/en/translator#' . esc_attr( $a['fromLang'] ) . '/' . esc_attr( $a['toLang'] ) . '/' . esc_attr( $content ) . '" style="font-size: larger; color: ' . esc_attr( $a['color'] ) . ';" target="_blank" title="Translate with DeepL">' . esc_attr( $content ) . " 🌐" . '</a></strong>';
        return $output;
}

add_shortcode('DeepL', 'denade_deepl_translate');


//   Google Translator 
function denade_google_translate($atts, $content) {
    $a = shortcode_atts( array(
        'color' => '#ed9f2d',
        'fromLang' => 'auto',
        'toLang' => 'en'
        ), $atts );
        $output = '<strong><a href="https://translate.google.com/?hl=en#view=home&op=translate&sl=' . esc_attr( $a['fromLang']) . '&tl=' . esc_attr( $a['toLang'] ) . '&text=' . esc_attr( $content ) . '" style="font-size: larger; color: ' . esc_attr( $a['color'] ) . ';" target="_blank" title="Translate with Google Translator">' . esc_attr( $content ) . " 🌐" . '</a></strong>';
        return $output;
}

add_shortcode('GTranslate', 'denade_google_translate');


//   Yandex Translate 
function denade_yandex_translate($atts, $content) {
    $a = shortcode_atts( array(
        'color' => '#ed9f2d',
        'fromLang' => 'de',
        'toLang' => 'en'
        ), $atts );
        $output = '<strong><a href="https://translate.yandex.com/?lang=' . esc_attr( $a['fromLang']) . '-' . esc_attr( $a['toLang'] ) . '&text=' . esc_attr( $content ) . '" style="font-size: larger; color: ' . esc_attr( $a['color'] ) . ';" target="_blank" title="Translate with Yandex.Translate">' . esc_attr( $content ) . " 🌐" . '</a></strong>';
        return $output;
}

add_shortcode('Yandex', 'denade_yandex_translate');


//   Baidu Translator 
function denade_baidu_translate($atts, $content) {
    $a = shortcode_atts( array(
        'color' => '#ed9f2d',
        'fromLang' => 'de',
        'toLang' => 'en'
        ), $atts );
        $output = '<strong><a href="https://fanyi.baidu.com/#' . esc_attr( $a['fromLang']) . '/' . esc_attr( $a['toLang'] ) . '/' . esc_attr( $content ) . '" style="font-size: larger; color: ' . esc_attr( $a['color'] ) . ';" target="_blank" title="Translate with Baidu">' . esc_attr( $content ) . " 🌐" . '</a></strong>';
        return $output;
}

add_shortcode('Baidu', 'denade_baidu_translate');
